src/providers/ses.rs

```diff
@@ -44,44 +46,70 @@ impl Provider for SesProvider {
         &self,
         to: &str,
         cc: &[&str],
+        headers: Option<&Headers>,
         subject: &str,
         body_text: &str,
         body_html: Option<&str>,
     ) -> Result<String, ProviderError> {
-        let mut destination = Destination::default();
-        destination.to_addresses = Some(vec![to.to_string()]);
+        let mut all_headers = header::Headers::new();
+        all_headers.set(header::From(vec![self.sender.parse().unwrap()]));
```

Comment:
Panicking inside a route handler is not acceptable I think. If one of these `parse` calls fails, we should fail gracefully and send an appropriate error response back to the caller.